### ISSUE TITLE
Dogbone Addin Menu item Missing 

### ISSUE THREAD

**Issue by 3DTek-xyz** [DVE2000/Dogbone#66]
HI
I have had an issue in the latest fusion on OSX.
The dogbone menu items are not shown at the bottom of the create DDL as they used to be.
Its available in manufacturing mode if the manufacturing model is edited, but not in the design workspace.
After allot of playing around I had the idea to set a keyboard shortcut through the manufacturing workspace, then flick back to design and try the key combo, and it works. So I can use the plugin again but... cannot seem to get the menu item back.

**Comment by pludikar** [comment]
Hi,

Hmm, I'll check if the same problem is happening on Windows. Typically, F360 has been OS agnostic, but I'll get back to you  shortly

Peter

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: 3DTek ***@***.***>
Sent: Wednesday, September 25, 2024 12:22:47 AM
To: DVE2000/Dogbone ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [DVE2000/Dogbone] Dogbone Addin Menu item Missing (Issue #66)


HI
I have had an issue in the latest fusion on OSX.
The dogbone menu items are not shown at the bottom of the create DDL as they used to be.
Its available in manufacturing mode if the manufacturing model is edited, but not in the design workspace.
After allot of playing around I had the idea to set a keyboard shortcut through the manufacturing workspace, then flick back to design and try the key combo, and it works. So I can use the plugin again but... cannot seem to get the menu item back.

**Comment by pludikar** [comment]
Hi,

I've updated the github files, but at the moment I have no idea if I've fixed anything in OSX.  I noticed that the resource image files were all wrong - there should be 3 sizes of image files 16x16, 32x32 and 64x64.  For some reason the files I had were completely off, and nowhere near the required sizes.  I know that OSX devices tend to be particular about file types and sizes, so this is just a guess at the moment.

Unfortunately It'll take me a while to get a Mac together to see if this has done anything, but if you could install the latest upload and let me know if it did anything, it would help me enormously.

Regards
Peter
________________________________

**Comment by 3DTek-xyz** [comment]
Hey Thanks for looking into this- I use the feature heavily and am glad to see it's so actively worked on and supported!
Ill give the new code a go - Thank again

**Comment by 3DTek-xyz** [comment]
Peter, sadly that didnt do it.  
In design workspace I can access the dogbone feature using my shorcut.
<img width="472" alt="image" src="https://github.com/user-attachments/assets/47eb4b32-b176-4261-91c0-9588aaa02e6e">
In Man workspace it shows up but without icon and without update available - not sure if that is a clue?
So I am unable to associate a shortcut to the update feature here in order to use in design mode.
<img width="360" alt="image" src="https://github.com/user-attachments/assets/9e4bd7ad-69ac-4ff8-b305-42775c05e8db">

Another note, for about 6 months I have noticed that the plugin needs restarting on every fusion opening, even though its listed to run on start.  It may be time for a full fusion uninstall / reinstall - as I expect someone else would have been running into these troubles and mentioning it if it was widespread.

**Comment by pludikar** [comment]
Thanks for checking it out.  OSX is a little bit harder for me to debug, but I do have access to a machine. I’ll look at it this in more detail later today.

I’ve noticed the failure to start too.  I do see an error message which only appears 1st time F360 fires up. I'll look into that too.

Peter

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by pludikar** [comment]
Hi,

I noticed that there is now an expanded set of resource files specifically for high resolution displays. I’ve added them to the latest GitHub file set.  Unfortunately, I’m having trouble getting hold of a suitable OSX machine (I mainly work Windows, for better or worse).

Would you mind checking the new file set out for me and let me know if these extra files have actually done anything. I’ve not seen any activity on the F360 forum, so I’m guessing it’s a software related issue related to the dogbone add-in and not OS related. I suspect OSX retina devices maybe looking for the image files I’ve just added.

I’m also working on the startup issue - looks like I’ll have to find a way around a f360 document internal validation error that only happens on initialization of F360, but once up and running the document becomes valid. This one may take a bit of thinking about.

Peter

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by 3DTek-xyz** [comment]
Peter,
It didnt help my issue of missing command buttons.  But Im now more convinced its my fusion, Im going to do the hard remove and reinstall.
Thanks for trying
Ben

**Comment by pludikar** [comment]
Hi Ben,

Please let me know if you have any success. I’m still trying to hunt down a suitable OSX machine.  Unfortunately, the only one I have is too old to run a compatible OSX version, but I will continue to persevere though.

The startup problem has the hallmarks of an f360 issue, although I still need to confirm it.

Peter

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by pludikar** [comment]
Hi Ben,

As a heads up - I've refactored the code and it looks like I've resolved the start-up issue - I've not merged the changes back into the main code trunk - you'll find it [here](https://github.com/DVE2000/Dogbone/tree/new-folder-structure) under "new-folder-structure" branch.

I still haven't resolved the initial issue of the command buttons appearing in the design environment, but the whole approach to creating the buttons has been changed to bring the code more into line with a F360 provided Add-in template.

As this is still on a git branch, there may be errors I've introduced - I will attempt to check it before I merge into the main trunk.

Peter
